alg/teca_tc_activity.py

```diff
@@ -51,93 +51,87 @@ def set_color_map(self, color_map):
+        generates a handful of histograms, summary statistics,
+        and plots. returns summary table with counts of annual
+        storms and their categories.
         """
-        def execute(port, data_in, req):
-            """
-            expects the output of the teca_tc_classify algorithm
-            generates a handful of histograms, summary statistics,
-            and plots. returns summary table with counts of annual
-            storms and their categories.
-            """
-            global plt
-            global plt_mp
-            global plt_tick
-
-            import matplotlib.pyplot as plt
-            import matplotlib.patches as plt_mp
-            import matplotlib.ticker as plt_tick
-
-            # store matplotlib state we modify
-            legend_frame_on_orig = plt.rcParams['legend.frameon']
-
-            # tweak matplotlib slightly
-            plt.rcParams['figure.max_open_warning'] = 0
-            plt.rcParams['legend.frameon'] = 1
-
-            # get the input table
-            in_table = teca_py.as_teca_table(data_in[0])
-            if in_table is None:
-                # TODO if this is part of a parallel pipeline then
-                # only rank 0 should report an error.
-                sys.stderr.write('ERROR: empty input, or not a table\n')
-                return teca_table.New()
-
-            time_units = in_table.get_time_units()
-
-            # get the columns of raw data
-            year = in_table.get_column('year').as_array()
-            region_id = in_table.get_column('region_id').as_array()
-            region_name = in_table.get_column('region_name')
-            region_long_name = in_table.get_column('region_long_name')
-            start_y = in_table.get_column('start_y').as_array()
-
-            ACE = in_table.get_column('ACE').as_array()
-            PDI = in_table.get_column('PDI').as_array()
-
-            # organize the data by year month etc...
-            regional_ACE = []
-            regional_PDI = []
-
-            # get unique for use as indices etc
-            uyear = sorted(set(year))
-            n_year = len(uyear)
-            ureg = sorted(set(zip(region_id, region_name, region_long_name)))
+        global plt
+        global plt_mp
+        global plt_tick
+
+        import matplotlib.pyplot as plt
+        import matplotlib.patches as plt_mp
+        import matplotlib.ticker as plt_tick
+
+        # store matplotlib state we modify
+        legend_frame_on_orig = plt.rcParams['legend.frameon']
+
+        # tweak matplotlib slightly
+        plt.rcParams['figure.max_open_warning'] = 0
+        plt.rcParams['legend.frameon'] = 1
+
+        # get the input table
+        in_table = teca_py.as_teca_table(data_in[0])
+        if in_table is None:
+            # TODO if this is part of a parallel pipeline then
+            # only rank 0 should report an error.
+            sys.stderr.write('ERROR: empty input, or not a table\n')
```

Comment:
This doesn't need to be addressed in this revision, but we should consider for future code/revisions that a more pythonic behavior would be for an exception to be raised (e.g., `raise RuntimeError("empty input, or not a table")`).